lib/galaxy/webapps/galaxy/api/configuration.py

```diff
@@ -2,28 +2,149 @@
+from . import (
+    get_admin_user,
+    get_app,
+    get_trans,
+    get_user,
+)
+from .common import (
+    parse_serialization_params,
+    SerializationKeysQueryParam,
+    SerializationViewQueryParam,
+)
 
 log = logging.getLogger(__name__)
 
+router = APIRouter(tags=['configuration'])
+
+AdminUserRequired = Depends(get_admin_user)
+
+
+def get_configuration_manager(app: StructuredApp = Depends(get_app)) -> ConfigurationManager:
+    return ConfigurationManager(app)
+
+
+@cbv(router)
+class FastAPIConfiguration:
+    configuration_manager: ConfigurationManager = Depends(get_configuration_manager)
+
+    @router.get(
+        '/api/whoami',
+        summary="Return information about the current authenticated user",
+        response_description="Information about the current authenticated user"
+    )
+    def whoami(self, user: User = Depends(get_user)) -> Optional[UserModel]:
+        """Return information about the current authenticated user."""
+        return _user_to_model(user)
+
+    @router.get(
+        '/api/configuration',
+        summary="Return an object containing exposable configuration settings",
+        response_description="Object containing exposable configuration settings"
+    )
+    def index(
+        self,
+        trans: ProvidesUserContext = Depends(get_trans),
+        view: Optional[str] = SerializationViewQueryParam,
+        keys: Optional[str] = SerializationKeysQueryParam,
+    ) -> Dict[str, Any]:
+        """
+        Return an object containing exposable configuration settings.
+
+        A more complete list is returned if the user is an admin.
+        Pass in `view` and a comma-seperated list of keys to control which
+        configuration settings are returned.
+        """
+        return _index(self.configuration_manager, trans, view, keys)
+
+    @router.get(
+        '/api/version',
+        summary="Return Galaxy version information: major/minor version, optional extra info",
+        response_description="Galaxy version information: major/minor version, optional extra info"
+    )
+    def version(self) -> Dict[str, Any]:
+        """Return Galaxy version information: major/minor version, optional extra info."""
+        return self.configuration_manager.version()
+
+    @router.get(
+        '/api/configuration/dynamic_tool_confs',
+        dependencies=[AdminUserRequired],
+        summary="Return dynamic tool configuration files",
+        response_description="Dynamic tool configuration files"
+    )
+    def dynamic_tool_confs(self) -> List[Dict[str, str]]:
+        """Return dynamic tool configuration files."""
+        return self.configuration_manager.dynamic_tool_confs()
+
+    @router.get(
+        '/api/configuration/decode/{encoded_id}',
+        dependencies=[AdminUserRequired],
+        summary="Decode a given id",
+        response_description="Decoded id"
+    )
+    def decode_id(
+        self,
+        trans: ProvidesAppContext = Depends(get_trans),
+        *,
```

Comment:
Is this `*` needed here?

Reply:
Yep. Otherwise you'll be a syntax error: non-default argument (encoded_id) follows default argument (trans). It's a common thing, I think it's mentioned somewhere in fastapi docs, but here's the PEP: https://www.python.org/dev/peps/pep-3102/#specification

Reply:
I think we need to move the id encoding and decoding into the id fields. I don't think trans should be needed to decode an id. I started looking into that and it would be a hige simplification for all serialization/deserialization. In the meantime, is there a problem moving trans to the last argument ? That seems like it'd be a more conventional setup.

Reply:
ConfigurationManager has a self.app - that can be used regardless of what cool type fixes we can do.

Reply:
^looking into this. Moving to WIP for now.

Reply:
Replaced str with EncodedDatabaseIdField. Adding one `typing.cast` fixed the typing issue, hurray! 
Also, specifying the encoded_id as a Query object solved the "non-default arg after default arg" error.
Now looking into dropping trans in the decode endpoint.

Thank you, everyone, for the detailed reviews!

Reply:
> I think we need to move the id encoding and decoding into the id fields. I don't think trans should be needed to decode an id. I started looking into that and it would be a hige simplification for all serialization/deserialization. In the meantime, is there a problem moving trans to the last argument ? That seems like it'd be a more conventional setup.

Done in #00b7ae8.